Dockerfile

```diff
@@ -1,68 +1,116 @@
-    && yum -y install proj-devel geos geos-devel spatialindex-devel p7zip p7zip-plugins freetype-devel libpng-devel wget git
+WORKDIR /tmp
 
-# install sqlite newer version
-RUN cd /tmp \
-    && curl https://sqlite.org/2024/sqlite-autoconf-3450100.tar.gz | tar xzf - \
+RUN dnf -y update \
+    && dnf -y groupinstall "Development Tools" \
+    && dnf -y install zlib-devel ncurses-devel gdbm-devel nss-devel openssl openssl-devel readline-devel libffi-devel \
+                      curl-devel bzip2-devel p7zip p7zip-plugins freetype-devel libpng-devel wget git \
+                      unzip cmake libtiff-devel sqlite-devel pkgconfig glibc-langpack-en
+
+# Set UTF-8 locale environment variables to ensure proper character encoding and avoid setlocale warnings
+ENV LANG=en_US.UTF-8
+ENV LANGUAGE=en_US:en
+ENV LC_ALL=en_US.UTF-8
+
+# Install geos from source (not in dnf)
+RUN curl -O https://download.osgeo.org/geos/geos-3.13.1.tar.bz2 \
+    && tar xvfj geos-3.13.1.tar.bz2 \
+    && cd geos-3.13.1 \
+    && mkdir _build \
+    && cd _build \
+    && cmake \
+        -DCMAKE_BUILD_TYPE=Release \
+        -DCMAKE_INSTALL_PREFIX=/usr/local \
+        .. \
+    && make \
+    && ctest \
+    && make install
+
+# Install libspatialindex from source (not in dnf)
+RUN wget https://github.com/libspatialindex/libspatialindex/releases/download/2.1.0/spatialindex-src-2.1.0.tar.bz2 \
+    && tar xvfj spatialindex-src-2.1.0.tar.bz2 \
+    && cd spatialindex-src-2.1.0 \
+    && mkdir build \
+    && cd build \
+    && cmake .. \
+    && make \
+    && make install \
+    && ldconfig
+
+# Install proj (not in dnf)
+RUN wget https://download.osgeo.org/proj/proj-9.6.2.tar.gz \
+    && tar xzf proj-9.6.2.tar.gz \
+    && cd proj-9.6.2 \
+    && mkdir build \
+    && cd build \
+    && cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/usr/local .. \
+    && cmake --build . \
+    && ctest \
+    && cmake --build . --target install \
+    && ldconfig
+
+RUN curl https://sqlite.org/2024/sqlite-autoconf-3450100.tar.gz | tar xzf - \
     && cd ./sqlite-autoconf-3450100 \
     && ./configure --prefix=/usr --libdir=/lib64 \
     && make \
-    && make install \
-    && rm -rf /tmp/sqlite-autoconf-3450100
+    && make install
 
+# Install AWS CLI v2
 RUN curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip" \
     && unzip awscliv2.zip  \
     && ./aws/install
 
-RUN cd /tmp \
-    && curl https://www.python.org/ftp/python/3.10.11/Python-3.10.11.tgz  | tar xzf - \
+# Install Python 3.10
+RUN curl https://www.python.org/ftp/python/3.10.11/Python-3.10.11.tgz | tar xzf - \
     && cd ./Python-3.10.11 \
     && ./configure --enable-optimizations --with-ensurepip=install \
     && make -j8 \
     && make altinstall \
-    && pip3.10 install pipenv virtualenv --upgrade \
-    && rm -r /tmp/Python-3.10.11
+    && pip3.10 install pipenv virtualenv --upgrade
```

Comment:
Is workers/dbloader the only thing using 3.11?  If so, maybe we dump 3.11 to keep the image size down and just bump that worker to 3.12.  I'm happy to do the work on dbloader.

Reply:
Ahh I forgot about pathstack.  That's fine then.